tests/test_noninteractive_mode.py

```diff
@@ -0,0 +1,80 @@
+""" Non-interactive tests. """
+import subprocess
+import csv
+from mssqltestutils import create_mssql_cli
+
+def test_session_closure_query_valid():
+    """ Test session closure for valid query. """
+    assert is_processed_closed("select 1")
+
+def test_session_closure_query_invalid():
+    """ Test session closure for invalid query. """
+    assert is_processed_closed("asdfasoifjas")
+
+def test_query_large():
+    query_str = "SELECT * FROM STRING_SPLIT(REPLICATE(CAST('X,' AS VARCHAR(MAX)), 1024), ',')"
+    p = subprocess.Popen("./mssql-cli -Q \"%s\"" % query_str, shell=True, stdin=subprocess.PIPE,
+                         stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    output = p.communicate(b'y\n')[0].decode("utf-8")
```

Comment:
> output = p.communicate(b'y\n')[0].decode("utf-8") [](start = 3, length = 50)

Doesn't this mean we're interactive?  If that's the case, we should make change to the product code so that we are non-interactive.

Reply:
This is only when the query exceeds a specified row limit—it prompts the user whether we should load everything. Is this considered interactive mode? If yes, is expected functionality to display everything regardless of query size?

Reply:
> This is only when the query exceeds a specified row limit—it prompts the user whether we should load everything. Is this considered interactive mode? If yes, is expected functionality to display everything regardless of query size?

yes, any prompt is considered interactive mode. You bring up a good point, what should the behavior be? We'll need to figure this out.

Reply:
I suppose we should display the VIM mode with no prompt—exact same functionality but we remove the prompt that shows up if rows exceed row limit. If you agree, I'll send out an update for this.

Reply:
I'd see what sqlcmd does. I'm not sure row limit makes sense in non-interactive mode.

Reply:
Just checked—SQLCMD outputs results without a prompt.